packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts

```diff
@@ -12,37 +12,31 @@ const {
  * Updates Cognito files that are tracked so that the diff is detected for an `amplify push`
  */
 export const updateCognitoTrackedFiles = async (): Promise<void> => {
+  if (await detectCognitoAttributesRequireVerificationBeforeUpdateDiff()) {
+    const { resourceName } = stateManager.getResourceFromMeta(stateManager.getMeta(), 'auth', 'Cognito', undefined, false)!;
+    await addExtraLineToCliInputsJson(pathManager.getBackendDirPath(), resourceName);
```

Comment:
what does this do? (apart from adding extra line, i.e. why adding line is important?)

Reply:
it enforces a resource change to be pushed